processor/src/it/java/com/linecorp/decaton/processor/CoreFunctionalityTest.java

```diff
@@ -103,4 +114,49 @@ public void testAsyncTaskCompletion() {
                 .build()
                 .run();
     }
+
+    /**
+     * Decaton doesn't guarantee that produced messages are processed in
```

Comment:
To avoid misleading, it would better to clarify state that
- this is not a part of spec
- however, we as developer can expect this behavior when setting concurrency to 1 so just checking that expectation here